### PR TITLE
Issue #917 - Update EndTacticalHealthMod() in XCGS_Unit to expose beta strike post mission healing to config

### LINKED ISSUE #917: Add Config option to disable Beta Strike's end tactical heal.
when beta strike is enabled, soldiers are healed for half of the damage they took when tactical ends. this prevents them from ever becoming gravely wounded and reduces recovery times. It'd be good to have an option to shut this feature off.

in XComGameState_Unit, function EndTacticalHealthMod()

### PR DESCRIPTION
**Pull request by BlackDog86**
Fixes #917

Added a boolean value to CHHelpers & updated code in EndTacticalHealthMod() in XCGS_Unit to allow other mods to set a config variable in XComGame.ini to disable the additional 50% health the game automatically grants to soldiers at the end of a mission when beta strike is active. 

 If the variable is false, or doesn't exist , the base game behaviour is retained.

### CODE REVIEW

**Comment by Iridar** [comment]
If the first message in the PR has something like "Closes #1192" or "Fixes #1192", then the corresponding issue will be closed automatically when the PR is merged, so do that in the future, please. I've edited the post for you for now.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 1]
```diff
@@ -2647,9 +2647,9 @@ function EndTacticalHealthMod()
 	local float HealthPercent, NewHealth, SWHeal;
 	local int RoundedNewHealth, HealthLost, NewMissingHP;
 
-	HealthLost = HighestHP - LowestHP;
-
-	if (LowestHP > 0 && `SecondWaveEnabled('BetaStrike'))  // Immediately Heal 1/2 Damage
+	HealthLost = HighestHP - LowestHP;	
+	//	Issue #917: Allow beta strike additional healing to be disabled by config
+	if (LowestHP > 0 && `SecondWaveEnabled('BetaStrike') && !class'CHHelpers'.default.bDisableBetaStrikePostMissionHealing)  // Immediately Heal 1/2 Damage
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:2652]: Nitpicking, but I think that for the sake of clarity it would be better to put in a separate if() statement with your config variable.

**Review by Iridar — CHANGES_REQUESTED** [review]

File: X2WOTCCommunityHighlander/Config/XComGame.ini [hunk 2]
```diff
@@ -256,3 +256,7 @@ bEnableVersionDisplay=true
 ; destroying smoke trails (and similar) prematurely.
 ; Values below 5 (seconds) will be ignored.
 ;ProjectileParticleSystemExpirationDefaultOverride=60
+
+; Start Issue #917
+;;; HL-Docs: ref:BetaStrikeEndTacticalHeal
+;bDisableBetaStrikePostMissionHealing=true ; Uncomment this to disable the wound reduction after a mission with Beta Strike enabled
```
[the remark this answers is not in the document]
Comment (BlackDog86) [reply X2WOTCCommunityHighlander/Config/XComGame.ini]: committed

File: X2WOTCCommunityHighlander/Config/XComGame.ini [hunk 3]
```diff
@@ -256,3 +256,7 @@ bEnableVersionDisplay=true
 ; destroying smoke trails (and similar) prematurely.
 ; Values below 5 (seconds) will be ignored.
 ;ProjectileParticleSystemExpirationDefaultOverride=60
+
+; Start Issue #917
+;;; HL-Docs: ref:BetaStrikeEndTacticalHeal
```
[the remark this answers is not in the document]
Comment (BlackDog86) [reply X2WOTCCommunityHighlander/Config/XComGame.ini]: committed

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/CHHelpers.uc [hunk 4]
```diff
@@ -996,4 +999,4 @@ static final function array<SoldierClassAbilityType> RebuildSoldierClassAbilityT
 
 	return AbilityTypes;
 }
-// End Issue #815
+// End Issue #815
```
[the remark this answers is not in the document]
Comment (BlackDog86) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/CHHelpers.uc]: fixed

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 5]
```diff
@@ -2647,13 +2647,16 @@ function EndTacticalHealthMod()
...
+	if (LowestHP > 0 && `SecondWaveEnabled('BetaStrike'))  // Immediately Heal 1/2 Damage 
 	{
-		SWHeal = FFloor( HealthLost / 2 );
-		LowestHP += SWHeal;
-		HealthLost -= SWHeal;
+		if  (!class'CHHelpers'.default.bDisableBetaStrikePostMissionHealing)
+			{
+			SWHeal = FFloor( HealthLost / 2 );
+			LowestHP += SWHeal;
+			HealthLost -= SWHeal;
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:2658]: Please add indentation here. Four spaces is the standard.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 6]
```diff
@@ -2647,13 +2647,16 @@ function EndTacticalHealthMod()
 	local float HealthPercent, NewHealth, SWHeal;
 	local int RoundedNewHealth, HealthLost, NewMissingHP;
 
-	HealthLost = HighestHP - LowestHP;
-
-	if (LowestHP > 0 && `SecondWaveEnabled('BetaStrike'))  // Immediately Heal 1/2 Damage
+	HealthLost = HighestHP - LowestHP;	
+	//	Issue #917: Allow beta strike additional healing to be disabled by config
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:2651]: Again, since this is the actual code change, I think 

;;; HL-Docs: feature:BetaStrikeEndTacticalHeal; Issue #917

tag is warranted here.

File: X2WOTCCommunityHighlander/Config/XComGame.ini [hunk 7]
```diff
@@ -256,3 +256,7 @@ bEnableVersionDisplay=true
 ; destroying smoke trails (and similar) prematurely.
 ; Values below 5 (seconds) will be ignored.
 ;ProjectileParticleSystemExpirationDefaultOverride=60
+
+;;; HL-Docs: ref:BetaStrikeEndTacticalHeal; Issue #917
+; Uncomment this to disable the wound reduction after a mission with Beta Strike enabled
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Config/XComGame.ini:261]: I would reword the comment slightly.

Something like: "With Beta Strike enabled, soldiers are normally immediately healed for 50% of their missing HP after a mission. Uncomment this to disable this behavior."

**Review by Iridar — CHANGES_REQUESTED** [review]

**Comment by BlackDog86** [comment]
Issues resolved - ready for review

**Review by Iridar — CHANGES_REQUESTED** [review]
There's still a formatting issue with HL-Docs.

A HL-Docs entry must look either like this:

`/// HL-Docs: feature:YourFeature; issue:1337;`

or like this:

`/// HL-Docs: feature:YourFeature; issue:1337; tags:tactical`

Tags are optional.
Either `///` or `;;;` are used depending on whether this is in script or in config.

Each feature must have at least one HL-Docs entry like this, where both feature name and issue number are specified.

All other HL-Docs entries should have only the feature name with the `ref` keyword:

`;;; HL-Docs: ref:YourFeature;`

All comments under HL-Docs entry that are also prefaced with `///` or `;;;` get pulled into the **[Highlander online documentation](https://x2communitycore.github.io/X2WOTCCommunityHighlander/)** automatically, where the entry with the `feature` and `issue` is used to generate the header, while all other entries with `ref` are appended to the page.

The only exception to this rule is the `ref:Bugfixes`, which looks like this:

HL-Docs: ref:Bugfixes; issue:644

And it has both a `ref` and an `issue` number.

**Comment by BlackDog86** [comment]
Please tell me it's right this time! Changed committed & pushed :)

**Comment by Iridar** [comment]
> Avoid trailing semicolons in the HL-Docs lines (and you should probably lowercase issue)

(c) sempai

**Comment by Iridar** [comment]
The docs builder still fails, sorry, looks like I was wrong and the `tags` keyword is mandatory.

Try it like this:

`/// HL-Docs: feature:BetaStrikeEndTacticalHeal; issue:917; tags:`

**Comment by BlackDog86** [comment]
OK last time! Please work!

**Review by Iridar — APPROVED** [review]

**Comment by Iridar** [comment]
Praise Johnny Lump!
